src/mem/alloc.rs

```diff
@@ -618,6 +622,49 @@ pub fn mem_allocate(size: usize) -> Result<VirtAddr, ()> {
     unsafe { ALLOCATOR.slab_alloc(size) }
 }
 
+/// Allocate 'size' bytes and sets the memory to zero
+pub fn mem_callocate(size: usize) -> Result<VirtAddr, ()> {
+    unsafe {
+        let result: Result<VirtAddr, ()> = mem_allocate(size);
+        if let Ok(va) = result {
+            core::intrinsics::write_bytes(va.as_mut_ptr::<u8>(), 0, size);
```

Comment:
we can use `memset` from here https://github.com/AMDESE/linux-svsm/blob/fceb4b107ed132367a795c56a079e96b58cd9584/src/util/util.rs#L115